fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java

```diff
@@ -661,36 +673,40 @@ public void prepareRelatedMVs(Set<Table> queryTables,
             return;
         }
 
+        List<Pair<MaterializedViewWrapper, MvUpdateInfo>> mvInfos =
+                Lists.newArrayListWithExpectedSize(mvWithPlanContexts.size());
         for (MaterializedViewWrapper wrapper : mvWithPlanContexts) {
             MaterializedView mv = wrapper.getMV();
-            MvPlanContext mvPlanContext = wrapper.getMvPlanContext();
             try {
                 // mv's partitions to refresh
                 MvUpdateInfo mvUpdateInfo = queryMaterializationContext.getOrInitMVTimelinessInfos(mv);
```

Comment:
This is also a heavy operation, can add it into the future too?

Reply:
> This is also a heavy operation, can add it into the future too?

getOrInitMVTimelinessInfos() has multiple implements, it may introduce more concurrent issues.
like `logMVPrepare` is not designed for multi-thread